npc/controller.go

```diff
@@ -86,6 +89,10 @@ func (npc *controller) AddPod(obj *coreapi.Pod) error {
 	npc.Lock()
 	defer npc.Unlock()
 
+	if npc.nodeName != "" && obj.Spec.NodeName != npc.nodeName {
```

Comment:
I'd suggest to factorise this condition and similar ones in this commit in a function to better show the intent & hide implementation details.
Maybe something like:
```
func (npc *controller) isRemotePod(pod *coreapi.Pod) bool {
    return npc.nodeName != "" && pod.Spec.NodeName != npc.nodeName
}
```

or maybe the opposite `isLocalPod` which we would then negate.